### ISSUE TITLE
`typename` doesn't work on basic types.

### ISSUE THREAD

**Issue by csilvers** [Khan/genqlient#130]
It seems like the `typename` directive only works on structs.  But it would be great if it worked on basic types like `string`, so you could do:
```
query x {
    # @genqlient(typename: "RenderType")
    renderType
}
```
where `renderType` is a string in the graphql schema, and the generated code would be:
```
type RenderType string
struct x {
    renderType RenderType
}
```

**Comment by csilvers** [comment]
I tried to implement this by modifying `convertDefinition`, but I don't think that will work: that function assumes that there's a 1-to-1 mapping from Go types to Graphql types, and that isn't necessarily true in this case.  So maybe this is more a feature request than a bug.

**Comment by csilvers** [comment]
OK, I think it is a bug in genqlient.  WIll file separately.

**Comment by csilvers** [comment]
I think this can't be implemented until #131 is resolved.  But once it is, this diff might be sufficient:
```
--- a/generate/convert.go
+++ b/generate/convert.go
@@ -289,7 +289,7 @@ func (g *generator) convertDefinition(
                }, err
        }
        goBuiltinName, ok := builtinTypes[def.Name]
-       if ok {
+       if ok { && options.TypeName == "" {
                return &goOpaqueType{GoRef: goBuiltinName, GraphQLName: def.Name
}, nil
        }
 
@@ -479,6 +479,12 @@ func (g *generator) convertDefinition(
                return g.addType(goType, goType.GoName, pos)
 
        case ast.Scalar:
+               if builtinTypes[def.Name] != "" {
+                       // In this case, the user asked for a custom Go type-nam
e
+                       // for a built-in type, e.g. `typename MyString string`.
+                       return &goOpaqueType{GoRef: name, GraphQLName: def.Name}
, nil
+               }
+
```

(plus changing the tests to test for it: I changed it liks this:
```
--- a/generate/testdata/queries/TypeNames.graphql
+++ b/generate/testdata/queries/TypeNames.graphql
@@ -3,7 +3,11 @@ query TypeNames {
   # @genqlient(typename: "User")
   user { id name }
   # @genqlient(typename: "Item")
-  randomItem { id name }
+  randomItem {
+     id
+     # @genqlient(typename: "NameType")
+     name
+  }
   # (ok to reuse the name as long as they match)
   # @genqlient(typename: "User")
   users { id name }
```
)
